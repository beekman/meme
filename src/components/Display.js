import React from 'react';
import PropTypes from 'prop-types';

const Display = ({ headerText, image, footerText }) => (
  <>
    <p>{ headerText }</p>
    <img src={ image } />
    <p>{ footerText }</p>
  </>
);

Display.propTypes = {
  headerText: PropTypes.string.isRequired,
  footerText: PropTypes.string.isRequired,
  image: PropTypes.string.isRequired,
  onChange: PropTypes.func.isRequired
};

export default Display;
