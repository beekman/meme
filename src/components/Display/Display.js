import React from 'react';
import PropTypes from 'prop-types';
import styles from '../App.css';

const Display = ({ headerText, image, footerText }) => (
  <div className={ styles.Display }>

    <img src={ image } />
    <p className={ styles.headerText }>{ headerText }</p>
    <p className={ styles.footerText }>{ footerText }</p>

  </div>
);

Display.propTypes = {
  headerText: PropTypes.string.isRequired,
  footerText: PropTypes.string.isRequired,
  image: PropTypes.string.isRequired
};

export default Display;
