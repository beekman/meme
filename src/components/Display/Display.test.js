import React from 'react';
import { shallow } from 'enzyme';
import Display from './Display';

describe('Display component', () => {
  it('matches a snapshot', () => {
    const wrapper = shallow(<Display headerText="Fry Day" footerText="Friday" image="https://imgflip.com/s/meme/Futurama-Fry.jpg" />);
    expect(wrapper).toMatchSnapshot();
  });
});
