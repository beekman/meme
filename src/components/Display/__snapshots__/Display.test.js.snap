// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Display component matches a snapshot 1`] = `
<div
  className="Display"
>
  <img
    src="https://imgflip.com/s/meme/Futurama-Fry.jpg"
  />
  <p
    className="headerText"
  >
    Fry Day
  </p>
  <p
    className="footerText"
  >
    Friday
  </p>
</div>
`;
